sound/soc/sof/loader.c

```diff
@@ -192,6 +192,52 @@ static int ext_man_get_dbg_abi_info(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+static int ext_man_get_config_data(struct snd_sof_dev *sdev,
+				   const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_config_data *config =
+		container_of(hdr, struct sof_ext_man_config_data, hdr);
+
+	/* save pointer to adsp config data to make it accessible from any place in code */
+	sdev->adsp_config = config;
```

Comment:
I think set up pointer here is not enough, I think we should parse each elem and call `snd_sof_dbg_memory_info_init`
```suggestion
	sdev->adsp_config = config;
	if (sdev->first_boot) {
		err = snd_sof_get_config_value(sdev, EXT_MAN_CONFIG_MEMORY_USAGE_SCAN, &memory_scan_en);
	        if (!err && memory_scan_en) {
		      snd_sof_dbg_memory_info_init(sdev);
		}
	}
```

Better to have a loop here to parse and add debug print to each value and token.

Reply:
changed to loop